metrics/error.global/error.go

```diff
@@ -0,0 +1,305 @@
+	truncateTimeout   time.Duration
+	lockWaitQuery     string
+	stop              bool
+	truncateErrPolicy *errors.TruncateErrorPolicy
+	metricType        byte
+	emitTotal         bool
+}
+
+// ErrorGlobal collects error summary information for the error.global domain.
+// https://dev.mysql.com/doc/refman/8.4/en/performance-schema-error-summary-tables.html
+type ErrorGlobal struct {
+	db *sql.DB
+	// --
+	options map[string]*errorLevelOptions
+}
+
+// Verify collector implements blip.Collector interface.
+var _ blip.Collector = &ErrorGlobal{}
+
+// NewErrorGlobal makes a new Table collector,
+func NewErrorGlobal(db *sql.DB) *ErrorGlobal {
+	return &ErrorGlobal{
+		db:      db,
+		options: make(map[string]*errorLevelOptions),
+	}
+}
+
+// Domain returns the Blip metric domain name (DOMAIN const).
+func (t *ErrorGlobal) Domain() string {
+	return DOMAIN
+}
+
+// Help returns the output for blip --print-domains.
+func (t *ErrorGlobal) Help() blip.CollectorHelp {
+	return blip.CollectorHelp{
+		Domain:      DOMAIN,
+		Description: "Errors Summary",
+		Options: map[string]blip.CollectorHelpOption{
+			OPT_ALL: {
+				Name:    OPT_ALL,
+				Desc:    "Collect all errors",
+				Default: "yes",
+				Values: map[string]string{
+					"yes":     "All errors (ignore metrics list)",
+					"no":      "Specified metrics",
+					"exclude": "Excludes specified metrics",
+				},
+			},
+			OPT_TOTAL: {
+				Name:    OPT_TOTAL,
+				Desc:    "Return the total number of errors raised.",
+				Default: "yes",
+				Values: map[string]string{
+					"yes":  "Return the total number of errors raised",
+					"no":   "Do not return the total number of errors raised",
+					"only": "Only return the total number of errors raised",
+				},
+			},
+			OPT_TRUNCATE_TABLE: {
+				Name:    OPT_TRUNCATE_TABLE,
+				Desc:    "If the source table should be truncated to reset data after each retrieval",
+				Default: "no",
+				Values: map[string]string{
+					"yes": "Truncate source table after each retrieval",
+					"no":  "Do not truncate source table after each retrieval",
+				},
+			},
+			OPT_TRUNCATE_TIMEOUT: {
+				Name:    OPT_TRUNCATE_TIMEOUT,
+				Desc:    "The amount of time to attempt to truncate the source table before timing out",
+				Default: "250ms",
+			},
+			OPT_TRUNCATE_ON_STARTUP: {
+				Name:    OPT_TRUNCATE_ON_STARTUP,
+				Desc:    "If the source table should be truncated on the start of metric collection. Truncation will use the timeout specified in " + OPT_TRUNCATE_TIMEOUT,
+				Default: "yes",
+				Values: map[string]string{
+					"yes": "Truncate source table on startup",
+					"no":  "Do not truncate source table on startup",
+				},
+			},
+		},
+		Groups: []blip.CollectorKeyValue{
+			{Key: GRP_ERR_NUMBER, Value: "the error number, or an empty string for a total"},
+			{Key: GRP_ERR_NAME, Value: "the error name, or an empty string for a total"},
+		},
+		Metrics: []blip.CollectorMetric{
+			{
+				Name: "raised",
+				Type: blip.DELTA_COUNTER,
```

Comment:
Since truncate=no by default, shouldn't this be cumulative counter by default?

Reply:
It is. This is a typo.